### ISSUE TITLE
Anti-alias on fill layer causes border to be drawn above other layers

### ISSUE THREAD

**Issue by stevage** [mapbox/mapbox-gl-js#9162]
I'm seeing this issue in Studio.

<img width="1190" alt="Screen Shot 2020-01-07 at 11 49 55 am" src="https://user-images.githubusercontent.com/678344/71859715-046cd580-3144-11ea-9306-980841f7182c.png">

All the polygons here are in the same layer, with their color determined by a property. The opacity of the 1px stroke is zero.

Expected: a result much like this (no borders separately from their fills):

<img width="861" alt="Screen Shot 2020-01-07 at 11 54 00 am" src="https://user-images.githubusercontent.com/678344/71859828-7b09d300-3144-11ea-8eb9-61d36f18bb2f.png">

I guess this is a known issue? Couldn't find it mentioned anywhere though.

**Comment by ansis** [comment]
I don't think this is a known issue. Would it be possible to share an example style that reproduces this issue?

**Comment by stevage** [comment]
I have made you a beautiful repro.

https://codepen.io/stevebennett/pen/xxbYrLz

![Screen Shot 2020-01-08 at 9 52 27 am](https://user-images.githubusercontent.com/678344/71936120-a9e38000-31fc-11ea-961a-eafbb49e00b3.png)

**Comment by Yuki-Zhao** [comment]
I also found this problem, how to solve?
